gradlew fails with: Failed to apply plugin [id 'elasticsearch.global-build-info']: Index 1 out of bounds for length 1

Today when trying to build and Elasticseach (master) from source, the build environment fails after typing `gradlew ...anytask...`:

```
> Failed to apply plugin [id 'elasticsearch.global-build-info']
   > Index 1 out of bounds for length 1
```

Exactly what happens can be seen with stack traces enabled:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at org.elasticsearch.gradle.info.GlobalBuildInfoPlugin.findOriginUrl(GlobalBuildInfoPlugin.java:406)
        at org.elasticsearch.gradle.info.GlobalBuildInfoPlugin.gitInfo(GlobalBuildInfoPlugin.java:381)
        at org.elasticsearch.gradle.info.GlobalBuildInfoPlugin.apply(GlobalBuildInfoPlugin.java:80)
        at org.elasticsearch.gradle.info.GlobalBuildInfoPlugin.apply(GlobalBuildInfoPlugin.java:48)
        at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
```

This part tries to find out the URL of the GIT repo by parsing the config file in `rootdir/.git/config` and looking for urls below "remote origin":

```
[remote "origin"]
	url = https://github.com/uschindler/elasticsearch.git
	fetch = +refs/heads/*:refs/remotes/origin/*
	puttykeyfile = 
[branch "master"]
```

After removing the "puttykeyfile =" line, it works. This settings was obviously added by my git client (TortoiseGit). According to the syntax of this file, this is perfectly valid.

`String#split()` in Java by default ignores empty tokens at end, so splitting on "=" retuns a one-element array. BTW, `String#split(String)` is a classical case for forbiddenapis, it's one of the most horrible methods in Java as it is completely unexpected what it's doing.

The fix is to add an if statement and ignore this line.

--- a/buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java
+++ b/buildSrc/src/main/java/org/elasticsearch/gradle/info/GlobalBuildInfoPlugin.java
@@ -402,7 +402,7 @@ public class GlobalBuildInfoPlugin implements Plugin<Project> {
                         // we're on to the next config item so stop looking
                         break;
                     }
-                    String[] pair = line.trim().split("=");
+                    String[] pair = line.trim().split("=", 2);
                     props.put(pair[0].trim(), pair[1].trim());
                 } else {
                     if (line.equals("[remote \"origin\"]")) {

